SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/TargetView.cpp

```diff
@@ -0,0 +1,99 @@
+{
+	Builder.AddEntity(EntityId);
+}
+
+void FTargetView::RemoveEntity(Worker_EntityId EntityId)
+{
+	// Make sure ops are generated to remove all authority and components for the removed entity.
+	EntityViewElement EntityData = View.FindAndRemoveChecked(EntityId);
+	for (const Worker_ComponentSetId& SetId : EntityData.Authority)
+	{
+		Builder.SetAuthority(EntityId, SetId, WORKER_AUTHORITY_NOT_AUTHORITATIVE, {});
+	}
+	for (const ComponentData& Component : EntityData.Components)
+	{
+		Builder.RemoveComponent(EntityId, Component.GetComponentId());
+	}
+	Builder.RemoveEntity(EntityId);
+}
+
+void FTargetView::AddOrSetComponent(Worker_EntityId EntityId, ComponentData Data)
+{
+	EntityViewElement* EntityData = View.Find(EntityId);
+	if (EntityData == nullptr)
+	{
+		Builder.AddEntity(EntityId);
+	}
+	ComponentData* Component = EntityData->Components.FindByPredicate(ComponentIdEquality{ Data.GetComponentId() });
+	if (Component != nullptr)
+	{
+		*Component = Data.DeepCopy();
+	}
+	else
+	{
+		EntityData->Components.Add(Data.DeepCopy());
+	}
+	Builder.AddComponent(EntityId, MoveTemp(Data));
+}
+
+void FTargetView::UpdateComponent(Worker_EntityId EntityId, ComponentUpdate Update)
+{
+	EntityViewElement& EntityData = View.FindChecked(EntityId);
+	ComponentData* Component = EntityData.Components.FindByPredicate(ComponentIdEquality{ Update.GetComponentId() });
+	check(Component != nullptr);
+	Component->ApplyUpdate(Update);
+	Builder.UpdateComponent(EntityId, MoveTemp(Update));
+}
+
+void FTargetView::RemoveComponent(Worker_EntityId EntityId, Worker_ComponentId ComponentId)
+{
+	EntityViewElement& EntityData = View.FindChecked(EntityId);
+	ComponentData* Component = EntityData.Components.FindByPredicate(ComponentIdEquality{ ComponentId });
+	check(Component != nullptr);
+	EntityData.Components.RemoveAtSwap(Component - EntityData.Components.GetData());
+	Builder.RemoveComponent(EntityId, ComponentId);
+}
+
+void FTargetView::AddAuthority(Worker_EntityId EntityId, Worker_ComponentSetId ComponentSetId)
+{
+	EntityViewElement* EntityData = View.Find(EntityId);
+	if (EntityData == nullptr)
+	{
+		Builder.AddEntity(EntityId);
+	}
+	check(!EntityData->Authority.Contains(ComponentSetId));
+	EntityData->Authority.Add(ComponentSetId);
+	Builder.SetAuthority(EntityId, ComponentSetId, WORKER_AUTHORITY_AUTHORITATIVE, {});
+}
+
+void FTargetView::RemoveAuthority(Worker_EntityId EntityId, Worker_ComponentSetId ComponentSetId)
+{
+	EntityViewElement& EntityData = View.FindChecked(EntityId);
+	const int32 Removed = EntityData.Authority.RemoveSingleSwap(ComponentSetId);
+	check(Removed);
+	Builder.SetAuthority(EntityId, ComponentSetId, WORKER_AUTHORITY_NOT_AUTHORITATIVE, {});
+}
+
+const EntityView& FTargetView::GetView() const
+{
+	return View;
+}
+
+OpList FTargetView::CreateOpListFromChanges()
+{
+	EntityComponentOpListBuilder Temp = MoveTemp(Builder);
+	Builder = EntityComponentOpListBuilder();
+	return MoveTemp(Temp).CreateOpList();
```

Comment:
What is the purpose of the MoveTemp(Temp) part?

Reply:
CreateOpList is a && func because when you call it you release resources to the caller. So forcing this makes it clear you are in a moved-from state and shouldn't use the object further